warcio/digestverifyingreader.py

```diff
@@ -91,16 +94,19 @@ def _update(self, buff):
             self.block_digester.update(buff)
 
         if self.limit == 0:
-            if _compare_digest_rfc_3548(self.block_digester, self.block_digest) is False:
-                self.problem('block digest failed: {}'.format(self.block_digest))
-            if _compare_digest_rfc_3548(self.payload_digester, self.payload_digest) is False:
-                self.problem('payload digest failed {}'.format(self.payload_digest))
+            check = _compare_digest_rfc_3548(self.block_digester, self.block_digest)
+            if check is False:
+                self.digest_checker.problem('block digest failed: {}'.format(self.block_digest))
+            elif check is True and self.digest_checker.status is not False:
+                self.digest_checker.status = True
+            check = _compare_digest_rfc_3548(self.payload_digester, self.payload_digest)
+            if check is False:
+                self.digest_checker.problem('payload digest failed {}'.format(self.payload_digest))
+            elif check is True and self.digest_checker.status is not False:
+                self.digest_checker.status = True
```

Comment:
Just a thought, perhaps instead of `status` it should be `passed` to indicate that the check has passed?
And instead of problem() should be fail() to indicate check has failed? Pretty minor, but might improve readability slightly.

Reply:
This is part of the public interface so yes, the name is important. I changed .status to .passed, but I kept problem() because I suspect that once this code ships we'll discover situations where we want to report a problem and not have it be a failure.